src/fielddef.ts

```diff
@@ -338,6 +339,18 @@ export function isScaleFieldDef<F>(channelDef: ChannelDef<F>): channelDef is Sca
   return !!channelDef && (!!channelDef['scale'] || !!channelDef['sort']);
 }
 
+export function isPositionFieldDef<F>(channelDef: ChannelDef<F>): channelDef is PositionFieldDef<F> {
+  return !!channelDef && (!!channelDef['axis'] || !!channelDef['stack'] || !!channelDef['impute']);
+}
+
+export function isMarkPropFieldDef<F>(channelDef: ChannelDef<F>): channelDef is MarkPropFieldDef<F> {
+  return !!channelDef && !!channelDef['legend'];
+}
+
+export function isTextFieldDef<F>(channelDef: ChannelDef<F>): channelDef is TextFieldDef<F> {
```

Comment:
Wouldn't it make more sense to check the channel and then get the field def (which should be a text field def). Otherwise, this test isn't very safe.

Reply:
That would require the whole encoding and would require a lot of refactor, so I think this is better.

For text fieldDef without format, it is behaving like a normal fielddef anyway.